embind-emit-tsd does not work with PTHREAD_POOL_SIZE

Please include the following in your bug report:

**Version of emscripten/emsdk:**
3.1.47


**Failing command line in full:**

```
emscripten$ em++  -sUSE_PTHREADS -sPROXY_TO_PTHREAD --embind-emit-tsd=foo --bind -sPTHREAD_POOL_SIZE=1 ./test/other/embind_tsgen.cpp
worker sent an error! undefined:undefined: Cannot find module '/tmp/emscripten_temp_2s4362f8/a.out.worker.js'

/tmp/emscripten_temp_2s4362f8/tsgen_a.out.js:156
      throw ex;
      ^
Error: Cannot find module '/tmp/emscripten_temp_2s4362f8/a.out.worker.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1026:15)
    at Function.Module._load (node:internal/modules/cjs/loader:871:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at MessagePort.<anonymous> (node:internal/main/worker_thread:197:24)
    at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:736:20)
    at MessagePort.exports.emitMessage (node:internal/per_context/messageport:23:28) {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

--- a/emcc.py
+++ b/emcc.py
@@ -3259,6 +3259,9 @@ def phase_embind_emit_tsd(options, in_wasm, wasm_target, memfile, js_syms):
   # Required function to trigger TS generation.
   settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$callRuntimeCallbacks']
   settings.EXPORT_ES6 = False
+  # Disable proxying and thread pooling so a worker is not automatically created.
+  settings.PROXY_TO_PTHREAD = False
+  settings.PTHREAD_POOL_SIZE = 0
   # Disable minify since the binaryen pass has not been run yet to change the
   # import names.
   settings.MINIFY_WASM_IMPORTED_MODULES = False

--- a/test/test_other.py
+++ b/test/test_other.py
@@ -2961,6 +2961,9 @@ int f() {
                   '--preload-file', 'fail.js',
                   '-O3',
                   '-msimd128',
+                  '-sUSE_PTHREADS',
+                  '-sPROXY_TO_PTHREAD',
+                  '-sPTHREAD_POOL_SIZE=1',
                   '-lembind', # Test duplicated link option.
                   ]
     self.run_process([EMCC, test_file('other/embind_tsgen.cpp'),

